frontend/src/app/products/buy/buy.component.ts

```diff
@@ -0,0 +1,111 @@
+import { Component, OnInit } from '@angular/core';
+import {ActivatedRoute, Router} from '@angular/router';
+import {ProductsService, IProduct} from '../products.service';
+import {UserService} from 'src/app/user/user.service';
+import * as core from '@angular/core';
+import {
+  FormControl,
+  ValidatorFn,
+  AbstractControl,
+  FormGroup,
+  ValidationErrors,
+  Validators,
+} from '@angular/forms';
+
+
+@Component({
+  selector: 'app-buy',
+  templateUrl: './buy.component.html',
+  styleUrls: ['./buy.component.css']
+})
+export class BuyComponent implements OnInit {
+  productId: string;
+  product: IProduct;
+  transactionPrice: number;
+
+
+  deliveryForm = new FormGroup({
```

Comment:
Not sure if this the right way to handle delivery addresses.
Maybe save it to the user (in profile edit section)
Here allow him to edit